docs/specification.md

```diff
@@ -748,79 +658,132 @@ Parameter (in Michelson): `list nat`.
 
-**get_pauser**
+In the following subsection we discuss and present design decisions regarding FA2 specifics.
 
-Parameter (in Michelson): `contract address`.
+## FA2 Specifics
 
-- Return current pauser address.
+FA2 provides a framework for defining permission policies, but does not require any particular policy.
+Moreover, it suggests an approach where permission logic is defined in a different contract and the token contract can dynamically change the policy.
+For the Stablecoin project we need to answer several questions:
 
-[FA2]: https://gitlab.com/tzip/tzip/-/blob/131b46dd89675bf030489ded9b0b3f5834b70eb6/proposals/tzip-12/tzip-12.md
+1. FA2 recommends the "transfer hook" design pattern, as opposed to implementing a monolithic contract.
+Are there any requirements whether the stablecoin contract should be monolithic or follow this pattern (i. e. have `set_transfer_hook`)?
+2. For a monolithic contract: which permission policy should we implement?
+Should the whole policy or at least some part of it be hardcoded?
+E. g. can we assume that self transfer will always be permitted and there will be no custom policy?
+What about other policies?
+3. For a modular (non-monolithic) contract: what exactly should be implemented by us and put into the repository?
+Should we implement any particular `transfer_hook`?
+If we should, the questions from (2) apply here.
+4. Should we implement any `owner_hook`?
+5. The FA2 standard does not specify who is permitted to update operators on behalf of the token owner.
+Who should be allowed to do it in this project?
 
-## Safelist update
+Overall, the response from TQ was that there are no strict requirements and we should pick the best approaches from the business logic.
+At the early development stage we propose the following:
+1. We will implement both options (starting from the monolithic one because it's simpler), measure gas costs and then will decide what is better.
+The "transfer hook" approach is recommended, but may be infeasible due to gas costs.
+2. We will implement a single permission policy (as part of the token contract or as a separate transfer_hook – that depends on the previous point).
+ + Self transfer is permitted as well as operator transfer.
+ + Owner transfer policy for both sender and receiver is `Optional_owner_hook`.
+ + No custom permission policy.
+3. We will not implement any `owner_hook`.
+The contract is usable without them because we stick to `Optional_owner_hook`.
 
-**set_safelist**
+Regarding `update_operators` we will stick to the following logic:
+1. In each `update_operator` item `owner` MUST be equal to `SENDER`.
+2. Each address can have arbitrary number of operators.
 
-Parameter (in Michelson): `option address`.
+One more uncertainty is related to the `token_metadata` entrypoint: whether metadata is constant or can be changed during contract's lifetime.
```

Comment:
If Token_metadata can be changed during the lifetime of the contract, enabling the contract to add support for more token_ids in course of its life, should an operator added using `All_tokens` get automatic access to any new tokens added during the contract's life?

Reply:
I am not saying about enabling more token_ids. We will always have only `token_id = 0`. Each token ID has its own metadata and I am talking only about metadata for `token_id = 0`. Do we want to change things like `name` or `extras` after origination? We assume that no.